api/src/identity-access-management/infrastructure/repositories/refresh-token.repository.js

```diff
@@ -0,0 +1,101 @@
+import { temporaryStorage } from '../../../shared/infrastructure/temporary-storage/index.js';
+import { PromiseUtils } from '../../../shared/infrastructure/utils/promise-utils.js';
+import { RefreshToken } from '../../domain/models/RefreshToken.js';
+
+// Refresh tokens are stored into 2 storage collections
+//
+// "refresh-tokens:" collection with the refresh token as key and token infos as value
+const refreshTokenTemporaryStorage = temporaryStorage.withPrefix('refresh-tokens:');
+// "user-refresh-tokens:" collection with the user ID as key and the list of refresh tokens of the user as value
+const userRefreshTokensTemporaryStorage = temporaryStorage.withPrefix('user-refresh-tokens:');
+
+/**
+ * @typedef {function} findByToken
+ * @param {Object} params
+ * @param {string} params.token
+ * @returns {Promise<RefreshToken>}
+ */
+async function findByToken({ token }) {
+  const data = await refreshTokenTemporaryStorage.get(token);
+  if (!data) return null;
+  return new RefreshToken({ ...data, value: token });
+}
+
+/**
+ * @typedef {function} findAllByUserId
+ * @param {Object} params
+ * @param {string|number} params.userId
+ * @returns {Promise<Array<RefreshToken>>}
+ */
+async function findAllByUserId({ userId }) {
+  const tokens = await userRefreshTokensTemporaryStorage.lrange(userId);
+  if (!tokens || tokens.length === 0) return [];
+
+  return PromiseUtils.mapSeries(tokens, async (token) => findByToken({ token }));
```

Comment:
Le choix de nommage suivant a été retenu et je m'en réjouis : 

```javascript
constructor({ userId, scope, source, value }) {
    this.userId = userId;
    this.scope = scope;
    this.source = source;
    this.value = value;
  }
```

J'aimerais qu'on poursuive l'effort de cohérence aux autres bouts du code de manière à différencier clairement les `RefreshTokens`, qui sont les objets métier, et les `tokenValues` qui sont les valeur sérialisées : 

```suggestion
async function findByTokenValue({ tokenValue }) {
  const data = await refreshTokenTemporaryStorage.get(tokenValue);
  if (!data) return null;
  return new RefreshToken({ ...data, value: tokenValue });
}

/**
 * @typedef {function} findAllByUserId
 * @param {Object} params
 * @param {string|number} params.userId
 * @returns {Promise<Array<RefreshToken>>}
 */
async function findAllByUserId({ userId }) {
  const tokenValues = await userRefreshTokensTemporaryStorage.lrange(userId);
  if (!tokenValues || tokenValues.length === 0) return [];

  return PromiseUtils.mapSeries(tokenValues, async (tokenValue) => findByTokenValue({ tokenValue }));
```